lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1099,15 +1104,54 @@ static bool DeserializeAllCompilerFlags(swift::CompilerInvocation &invocation,
 
       found_swift_modules = true;
       StringRef moduleData = buf.substr(0, info.bytes);
-      got_serialized_options |=
-          DeserializeCompilerFlags(invocation, moduleData, info.name, error);
+
+      auto remap = [&](std::string path) {
+        ConstString remapped;
+        if (module.GetSourceMappingList().RemapPath(ConstString(path),
+                                                    remapped))
+          return remapped.GetStringRef().str();
+        return path;
+      };
+
+      /// Initialize the compiler invocation with it the search paths from a
+      /// serialized AST.
+      auto deserializeCompilerFlags = [&]() -> bool {
+        auto result = invocation.loadFromSerializedAST(moduleData);
+        if (result == swift::serialization::Status::Valid) {
+          if (discover_implicit_search_paths) {
```

Comment:
should this condition be moved before the call to `loadFromSerializedAST`? If `discover_implicit_search_paths` is false, are there desirable side effects to still calling `loadFromSerializedAST`?

Reply:
Having the condition in this location preserves the behavior we already had before this patch. We were previously loading things like the SDK path, triple, and curiously Clang import options into the invocation.